search/median_search2.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file median_search2.cpp
+ *
+ * @details
+ * Given a linked list L[0,....,n] of n numbers, find the middle node.
+ *
+ * Here are some example lists you can use to see how the algorithm works
+ * A = [1,2,3,4,5]
+ * B = [1,2,3,4,5,6]
+ * print median(A) #should be 39
+ * print median(B) #should be 4
+ *
+ * @author [Benjamin Weiss](https://github.com/weiss-ben)
+ */
+
+/**
+ * Definition for singly-linked list.
+ */
+struct ListNode {
+    int val;
+    ListNode *next;
+    ListNode() : val(0), next(nullptr) {}
+    ListNode(int x) : val(x), next(nullptr) {}
+    ListNode(int x, ListNode *next) : val(x), next(next) {}
+ };
+ 
+ #include <iostream>
+ #include <cassert>
+
+/**
+ * @namespace search
+ * @brief Search algorithms
+ */
+namespace search {
+/**
+ * @namespace median_search
+ */
+namespace median_search2 {
+/**
+* This function searches for the median of a linked list.
+* @param head, The head of the linked list.
+* @returns Median node of the linked list.
+*/  
+ListNode* middleNode(ListNode* head) {
+    if(!head)
+        return nullptr;
+        
+    //Fast and slow pointers
+    ListNode* fastptr;
+    ListNode* slowptr = fastptr = head;
+    
+	// fast jumps 2 while slow jumps 1    
+    while(fastptr->next && fastptr->next->next) {
+        slowptr = slowptr->next;
+        fastptr = fastptr->next->next;
+    }
+    return (fastptr->next) ? slowptr->next : slowptr;
+        
+}
+}  // namespace median_search2
+}  // namespace search
+
+/**
+ * Function to test above algorithm
+ */
+void test(){
```

Comment:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
 */
static void test() {
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable